PhysicsTools/NanoAOD/python/genparticles_cff.py

```diff
@@ -81,7 +81,68 @@
     )
 )
 
+finalGenParticlesWmass = cms.EDProducer("GenParticlePruner",
+    src = cms.InputTag("prunedGenParticles"),
+    select = cms.vstring(
+    "drop *",
+    "+keep abs(pdgId) == 11 || abs(pdgId) == 13 || abs(pdgId) == 15", #keep leptons, with at most one mother back in the history
+    "keep abs(pdgId) == 12 || abs(pdgId) == 14 || abs(pdgId) == 16",   # keep neutrinos     
+   )
+)
+
+
+
+##################### Tables for final output and docs ##########################
+genParticleTableWmass = cms.EDProducer("SimpleCandidateFlatTableProducer",
+    src = cms.InputTag("finalGenParticlesWmass"),
+    cut = cms.string(""), #we should not filter after pruning
+    name= cms.string("GenPartWmass"),
+    doc = cms.string("Charged leptons and neutrinos from GenParticles at high numerical precision"),
+    singleton = cms.bool(False), # the number of entries is variable
+    extension = cms.bool(False), # this is the main table for the taus
+    variables = cms.PSet(
+         pt  = Var("pt",  float),
```

Comment:
precision specifiers are missing for all these variables, but we should move to the model of per-particle precision enabled by the new feature, right? so I guess this table is all to be dropped and the per-particle precision be defined in the other table according to your needs